mypy/fastparse.py

```diff
@@ -905,7 +905,7 @@ def do_func_def(
                         self.errors, line=lineno, override_column=n.col_offset
                     ).translate_expr_list(func_type_ast.argtypes)
                     arg_types = [
-                        a if a is not None else AnyType(TypeOfAny.unannotated)
+                        a if a is not None else AnyType(TypeOfAny.unannotated)  # type: ignore[redundant-expr]
```

Comment:
Why the ignore?

Reply:
I ran into annoying errors due to `list` invariance. But I think I've found a workaround (see 4def89f)

Reply:
Actually I changed my mind about 4def86f66aea836b64f908c460678366299a1906; I think 63815f00f66f20c0cf947e03a70e47324c5cb988 is a better solution to the problem